daemon/monitor.go

```diff
@@ -0,0 +1,119 @@
+package daemon
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/libcontainerd"
+	"github.com/docker/docker/runconfig"
+)
+
+// StateChanged updates daemon state changes from containerd
+func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
+	c := daemon.containers.Get(id)
+	if c == nil {
+		return fmt.Errorf("no such container: %s", id)
+	}
+
+	switch e.State {
+	case libcontainerd.StateExit:
+		c.Reset(true)
+		daemon.Cleanup(c)
+		c.SetStopped(&container.ExitStatus{
+			ExitCode:  int(e.ExitCode),
+			OOMKilled: e.OOMKilled,
+		})
+		attributes := map[string]string{
+			"exitCode": fmt.Sprintf("%d", e.ExitCode),
```

Comment:
nitpick: could use `strconv.Itoa` or `strconv.FormatInt` to avoid format string.